Assets/Scripts/Infrastructure/PubSub/MessageChannel.cs

```diff
@@ -8,11 +8,10 @@ public class MessageChannel<T> : IMessageChannel<T>
     {

```

Comment:
What in-game issue is this fixing?
If we need to maintain this code, might was well include a 3rd party messaging library instead.
If we keep this, we need to add tests and release notes for this.

Reply:
In our current develop, we don't have actual in-game issues with it because of the way we use message channels. We had an issue with a previous iteration of #666 though, when subscribing and unsubscribing to a channel in a state's OnEnter and OnExit events. The latest version doesn't cause this issue anymore but I still think that this is a fix we should do, so that we don't end up having to re-do this whole investigation later when we encounter this issue again.

Reply:
@pdeschain we talked about how if we had to maintain this code, we'd look into a 3rd party. I consider the above maintaining. What would it take to build networked message publishing with a 3rd party lib?

Reply:
As discussed offline, there's work to maintain in both cases.
nvm, carry on